curve_checks.c

```diff
@@ -0,0 +1,111 @@
+// Mina elliptic cruve unit tests
+//
+//    Imported from https://github.com/MinaProtocol/c-reference-signer/blob/master/curve_checks.c
+
+#include "curve_checks.h"
+
+bool curve_checks(void)
+{
+    Affine a3;
+    Affine a4;
+    union {
```

Comment:
a mix of :grimacing: and :joy:

Reply:
Yeah - this was the first time I've used union in a long time.  Simply put, we were out of stack space and I had no choice! :)